### ISSUE TITLE
A few comments about https://my.numworks.com/apps

### ISSUE THREAD

**Issue by adriweb** [numworks/epsilon-sample-app-cpp#22]
Just trying some things here... :)

- The dynamically loaded WASM files served on the page don't have the expected content-type (`0a7f568c.js:1 wasm streaming compile failed: TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.`
- We have to open the browser console in order to see link issues (things like `undefined reference to 'pow'`) - could they be presented somewhere on screen?
- `Size: NaN KB` is displayed when linking fails
- Not sure what to do about this showing up in the console...? `0a7f568c.js:1 /Volumes/data/jenkins/workspace/GNU-toolchain/arm-11/src/newlib-cygwin/libgloss/libnosys/sbrk.c:21: undefined reference to 'end'`.
**_Edit 1:_** Locally, adding a `end = .;` in the `heap` part of the linker script seems to solve that, but... I'm not sure if this is actually a correct fix, and it doesn't solve the problem when using the website anyway.
**_Edit 2:_** I'm still not sure if it's a correct fix, but to make it work on the website (not touching the linker script), I added a global `void* end;` and `end = _heap_start;` in `_eadk_main`.

**Comment by Ecco** [comment]
> We have to open the browser console in order to see link issues (things like undefined reference to 'pow') - could they be presented somewhere on screen?

Hmm, at the moment this is low priority because there should *not* be any error to begin with. Indeed, you most *users* would not be able to do anything with those errors anyway.

Latest releases of `nwlink` and this sample app will run the final link step locally and give *developers* those error messages. This should fix the root problem!

> Size: NaN KB is displayed when linking fails

Same. There should not be any linking error. Of course ideally we should handle them gracefully, but this is low priority 😄 

> Not sure what to do about this showing up in the console...? 0a7f568c.js:1 /Volumes/data/jenkins/workspace/GNU-toolchain/arm-11/src/newlib-cygwin/libgloss/libnosys/sbrk.c:21: undefined reference to 'end'.

I have no idea what's going on there.

**Comment by Miniontoby** [comment]
I got this error in my console logs when trying to upload the released file (from the releases tab) to the /apps page: 
```
Uncaught (in promise) ./this.program: Missing api level. Please define a .rodata.eadk_api_level section.
./this.program: eadk.o: in function `_start':
(.text._start+0x1c): undefined reference to `main'
```

**Comment by Ecco** [comment]
@Miniontoby Indeed, we forgot to update the released file. That's fixed, version 1.2 works correctly.

**Comment by Ecco** [comment]
Also, I think the issues that were initially reported here are now all fixed, so I'm going to close this issue. Feel free to create new ones if needed!
